### PR TITLE
Add ability to parse dates and metadata tags from posting comments

### PR DESCRIPTION
**Pull request by rumpuslabs**
Ledger comments can contain metadata, including posting-specific dates:
```
  My::Account   $5.00   ; [2024-01-31]
```
and tags, optionally with values of various types:
```
  My::Account   $5.00   ; :Tag1:Tag2:
                        ; StringTag: string value
                        ; IntTag:: 1234
                        ; FloatTag:: 3.141592
                        ; DateTag:: [2024-01-31]
```

This PR adds support for parsing these values and including them in the Posting struct.

If the comment appears between the transaction header and the first posting, they instead appear in the Transaction struct. Ledger's documentation says that these comments are applied to all postings in the transaction but that is out of scope of this crate which only handles parsing.

Because fields have been added to the Transaction and Posting structs, I have bumped the version to 6.0.0; however most users of this crate will not be constructing these structs themselves or matching exhaustively on the struct fields so I imagine most existing code will work without changes.

### CODE REVIEW

File: src/model.rs [hunk 1]
```diff
@@ -48,12 +65,15 @@ impl fmt::Display for LedgerItem {
...
     pub code: Option<String>,
     pub description: String,
+    pub comment: Option<String>,
+    pub date: NaiveDate,
+    pub effective_date: Option<NaiveDate>,
+    pub posting_date: Option<NaiveDate>,
+    pub posting_effective_date: Option<NaiveDate>,
+    pub tags: Vec<Tag>,
     pub postings: Vec<Posting>,
 }
```
Comment (marek-g) [inline src/model.rs]: This is correct, of course, but there's a lot of the same fields added for Transaction and Posting. Also it is not very clear where do they came from.

What do you think about embedding `PostingMetadata` struct for `Transaction` and for the `Posting`? Especially if new fields are not belonging to the transaction but to all the postings (if they are not properties of Transaction there's no need to clutter top level).

```rust
#[derive(Debug, PartialEq, Eq, Clone)]
pub struct Transaction {
    pub date: NaiveDate,
    pub effective_date: Option<NaiveDate>,
    pub status: Option<TransactionStatus>,
    pub code: Option<String>,
    pub description: String,
    pub comment: Option<String>,
    pub postings: Vec<Posting>,
    pub postings_metadata: PostingMetadata,
}

#[derive(Debug, PartialEq, Eq, Clone)]
pub struct Posting {
    pub account: String,
    pub reality: Reality,
    pub amount: Option<PostingAmount>,
    pub balance: Option<Balance>,
    pub status: Option<TransactionStatus>,
    pub comment: Option<String>,
    pub metadata: PostingMetadata,
}

#[derive(Debug, PartialEq, Eq, Clone)]
pub struct PostingMetadata {
    date: Option<NaiveDate>,
    effective_date: Option<NaiveDate>,
    tags: Vec<Tag>,
}
```

I slightly prefer that comments are not part of the `PostingMetadata` - it's like something different for me (the current `Metadata` struct could stay as it is to not make too many changes in the parser code) and makes the changes more compatible with previous versions, but if you see it as being cleaner to have comments inside `PostingMetadata` (or called just `Metadata`) - I'm fine with it.

What do you think?

**Review by marek-g — CHANGES_REQUESTED** [review]
Thanks Tim! That's a great PR. I only would like to talk what do you think about my proposition of grouping metadata fields (see my comment).

File: src/model.rs (hunk 1 above)
Comment (rumpuslabs) [reply src/model.rs]: Done as suggested. :+1:

Comment (marek-g) [reply src/model.rs]: That's great! Thank you Tim!

**Pull request merged**